mypy/partially_defined.py

```diff
@@ -507,7 +507,8 @@ def visit_break_stmt(self, o: BreakStmt) -> None:
         self.tracker.skip_branch()
 
     def visit_expression_stmt(self, o: ExpressionStmt) -> None:
-        if isinstance(self.type_map.get(o.expr, None), (UninhabitedType, type(None))):
+        typ = self.type_map.get(o.expr)
+        if typ is None or isinstance(get_proper_type(typ), UninhabitedType):
```

Comment:
Can this be tested?

Reply:
I.e., can there be a test that would have failed without this change but now passes? It seems like that would have to require a non-"proper" type that resolves to UninhabitedType.